driver-async/src/main/com/mongodb/async/client/internal/KeyManagementService.java

```diff
@@ -71,8 +69,7 @@ public void completed(final Void aVoid) {
             @Override
             public void failed(final Throwable t) {
                 stream.close();
-                callback.onResult(null, new MongoSocketOpenException("Exception opening connection to Key Management Service",
-                        getServerAddress(keyDecryptor), t));
+                callback.onResult(null, t instanceof MongoSocketException ? t.getCause() : t);
```

Comment:
The next two changes make this async version throw the same exceptions as the sync version.  The issue is that the async version uses a `Stream`, which already wraps `IOException` in a `MongoSocketException`.  But we don't really want to throw a `MongoSocketException` here because we're not actually talking to MongoDB.  By unwrapping the `MongoSocketException`, we get the cause (which will be an `IOException`) and return it, and let the caller wrap it in a `MongoClientException`

Reply:
Good idea.  Done.